lib/Runtime/Library/JavascriptProxy.cpp

```diff
@@ -919,11 +912,54 @@ namespace Js
         Var trapResult = getEnumeratorMethod->CallFunction(arguments);
         threadContext->SetImplicitCallFlags((Js::ImplicitCallFlags)(saveImplicitCallFlags | ImplicitCall_Accessor));
 
-        if (!JavascriptOperators::IsObject(trapResult))
+        if (!JavascriptOperators::IsArray(trapResult))
```

Comment:
we should leave this to the ownKeys trap instead of having a separate invocation path here.